### PR TITLE
Fix discriminator key descriptor problem closes #38

### LINKED ISSUE #34: Ensure Mongoose 5 compatibility
In [our gitter channel](https://gitter.im/RealMQ/node-mongo-tenant?at=5aa5a47d8f1c77ef3aac02f6) the question came up if mongo-tenant is compatible with mongoose 5.

see [migration guide](https://github.com/Automattic/mongoose/blob/master/migrating_to_5.md)

* [x] Make sure mongo-tenant is compatible with the latest mongoose major version.
* [x] Adjust tests
* [x] Adjust readme

### LINKED ISSUE #38: Problem with discriminator key
Hello, I'm trying to use mongo-tenant with [mongoose discriminators](http://mongoosejs.com/docs/discriminators.html). But I'm having a problem in this piece of the code:

```js
// inherit all static properties from the mongoose base model
    for (let staticProperty of Object.getOwnPropertyNames(BaseModel)) {
      if (MongoTenantModel.hasOwnProperty(staticProperty)
      || ['arguments', 'caller'].indexOf(staticProperty) !== -1
      ) {
        continue;
      }

      MongoTenantModel[staticProperty] = BaseModel[staticProperty];
    }

    return MongoTenantModel;
```

This throws the following error:

```
TypeError: Cannot assign to read only property 'baseModelName' of function 'class MongoTenantModel extends BaseModel
```

The problem is that the discriminator sets a property `baseModelName` that has the following descriptor:

```js
{ 
  value: 'baseModelName',
  writable: false,
  enumerable: false,
  configurable: true
}
```

And since is not settable, it throws an error.

### PR DESCRIPTION
**Pull request by felipe-augusto**
Fixes the discriminator key problem #38, where the property is not writable.

### CODE REVIEW

**Comment by alrik** [comment]
Hey @felipe-augusto,

thanks a lot for fixing this issue.
Could you add a test case?

Cheers :)

File: index.js [hunk 1]
```diff
@@ -311,7 +311,8 @@ class MongoTenant {
         continue;
       }
 
-      MongoTenantModel[staticProperty] = BaseModel[staticProperty];
+      let descriptor = Object.getOwnPropertyDescriptor(BaseModel, staticProperty)
```
Comment (alrik) [inline index.js]: Please add a semicolon to the end of line, so that the codebase stays consistent. Thx

**Review by alrik — CHANGES_REQUESTED** [review]

**Comment by felipe-augusto** [comment]
I don't know if the test I wrote is in the right place, can you confirm it @alrik?
Also, I'm not sure why this didn't passed on Travis.

**Review by alrik — APPROVED** [review]

**Comment by alrik** [comment]
Thanks @felipe-augusto,
looks good to me.

Yes, there are some strage timeouts with travis :(
I restarted the pipeline and everything is fine :+1:

**Pull request merged**